grove/measurements/estimation.py

```diff
@@ -0,0 +1,161 @@
+"""
+Utilities for estimating expected values of Pauli terms given pyquil programs
+"""
+import numpy as np
+from pyquil.paulis import PauliSum, PauliTerm, commuting_sets, sI
+from pyquil.quil import Program
+from pyquil.gates import RY, RX
+
+
+class CommutationError(Exception):
+    pass
+
+
+def remove_imaginary_terms(pauli_sums):
```

Comment:
Can you comment on the use cases? This is clearly not a projection to the hermitian component of a Pauli sum but more like filtering the terms...

Reply:
This method is a little out of scope of the PR but I had it in some scrap code associated with an earlier version of the estimation routines.  When I want to measure a linear combination of Pauli terms that are generated from some fermion-to-qubit transform I know my expected value is going to be real.  Therefore, I know the expected value of all the imaginary components will cancel out so I don't need to spend the measurement effort to evaluate them.  The change you requested earlier meshes with this still.